packages/blocks/src/__internal__/utils/selection.ts

```diff
@@ -870,29 +875,59 @@ export function restoreSelection(selectedBlocks: SelectedBlock[]) {
 }
 
 /**
- * Get the closest editor element in the horizontal position
+ * Get the closest element in the horizontal position
  */
-export function getClosestHorizontalEditor(clientY: number) {
+export function getHorizontalClosestElement<
```

Comment:
I borrowed this function from #886. Don't worry, I will resolve the conflict after.